pkg/kudoctl/cmd/install/install.go

```diff
@@ -0,0 +1,284 @@
+package install
+
+import (
+	"fmt"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/check"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/github"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/helpers"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/k8s"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/vars"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	"strings"
+)
+
+func InstallCmd(cmd *cobra.Command, args []string) error {
+	// Validating flags
+	/*
+		// this makes --frameworkname mandatory
+		frameworkNameFlag, err := cmd.Flags().GetString("frameworkname")
+		if err != nil || frameworkNameFlag == "" {
+			return fmt.Errorf("Please set Frameworkname flag, e.g. \"--frameworkname=kafka\"")
+		}
+	*/
+	_, err := cmd.Flags().GetString("kubeconfig")
+	// This makes --kubeconfig flag optional
+	if err != nil {
+		return fmt.Errorf("get flag: %+v", err)
+	}
+
+	err = check.KubeConfigPath()
+	if err != nil {
+		return errors.WithMessage(err, "could not check kubeconfig path")
+	}
+
+	err = installFrameworks(args)
+	if err != nil {
+		return errors.WithMessage(err, "could not install framework(s)")
+	}
+
+	return nil
+}
+
+func installFrameworks(args []string) error {
+
+	if len(args) < 1 {
```

Comment:
I was contemplating checking e.g. `args` above and for a while indifferent but then found it more suiting to move it in its own function that holds the install logic. For unit testing that also meant I was able to logically separate  `args` from `flags` and the `kubeconfig` so that I could only focus on the installation part.